cypress/integration/PopupStack.spec.ts

```diff
@@ -0,0 +1,153 @@
+        .should(beOnTopOfLabelledByText('Window 4'));
+    });
+  });
+
+  context('when Delete Item button is clicked', () => {
+    beforeEach(() => {
+      cy.contains('button', 'Delete Item').click();
+    });
+
+    it('should open Delete Item popup', () => {
+      cy.findByLabelText('Delete Item').should('be.visible');
+    });
+
+    context('when Window 2 is clicked', () => {
+      beforeEach(() => {
+        cy.findByLabelText('Window 2').click();
+      });
+
+      it('should close Delete Item popup', () => {
+        cy.findByLabelText('Delete Item').should('not.exist');
+      });
+
+      it('should place Window 2 above others', () => {
+        cy.findByLabelText('Window 2')
+          .should(beOnTopOfLabelledByText('Window 1'))
+          .should(beOnTopOfLabelledByText('Window 3'));
+      });
+    });
+
+    context('when Window 2 Tooltip is hovered', () => {
+      beforeEach(() => {
+        cy.findByText('Contents of Window 2').trigger('mouseover');
+      });
+
+      context('when Window 2 Tooltip is clicked', () => {
+        beforeEach(() => {
+          cy.findByText('Contents of Window 2').click();
+        });
+
+        it('should close Delete Item popup', () => {
+          cy.findByLabelText('Delete Item').should('not.exist');
+        });
+
+        it('should place Window 2 above others', () => {
+          cy.findByLabelText('Window 2')
+            .should(beOnTopOfLabelledByText('Window 1'))
+            .should(beOnTopOfLabelledByText('Window 3'));
+        });
+      });
+
+      context('when the Escape key is pressed', () => {
+        beforeEach(() => {
+          cy.get('html').trigger('keydown', {
+            key: 'Escape',
+          });
+        });
+
+        it('should close the Tooltip', () => {
+          cy.findByRole('tooltip').should('not.exist');
+        });
+
+        it('should not close the Delete Item popup', () => {
+          cy.findByLabelText('Delete Item').should('be.visible');
+        });
+
+        context('when the Escape key is pressed again', () => {
+          beforeEach(() => {
+            cy.get('html').trigger('keydown', {
+              key: 'Escape',
+            });
+          });
+
+          it('should close the Delete Item popup', () => {
+            cy.findByLabelText('Delete Item').should('not.exist');
+          });
+        });
+      });
+    });
+
+    context('when user clicks outside', () => {
+      beforeEach(() => {
+        cy.get('html').click('bottomRight');
+      });
+
+      it('should close Delete Item popup', () => {
+        cy.findByLabelText('Delete Item').should('not.exist');
+      });
+    });
+
+    context('when the Escape key is pressed', () => {
+      beforeEach(() => {
+        cy.get('html').trigger('keydown', {
+          key: 'Escape',
+        });
+      });
+
+      it('should close Delete Item popup', () => {
+        cy.findByLabelText('Delete Item').should('not.exist');
+      });
+    });
+  });
+});
```

Comment:
Not all of these behaviors have been fully verified by accessibility and UX. I figure that will be a much longer conversation involving planning. Canvas Kit doesn't have the concept of a `Window` anyways, so I thought I'd put up an initial specification to get things started. Changes to some of these specs will not break anything since `Window` doesn't exist yet.